api/nodes/views.py

```diff
@@ -357,11 +342,14 @@ def get_object(self):
     def perform_destroy(self, instance):
         auth = get_user_auth(self.request)
         node = self.get_object()
+
+        if enforce_no_children(self.request) and Node.objects.get_children(node, active=True).exists():
+                raise ValidationError('Any child components must be deleted prior to deleting this project.')
```

Comment:
DELETE /v2/nodes/<node_id> versioned so prior to v2.12, children must be deleted first.  For 2.12 and greater, entire hierarchy is deleted, provided you have perms.